tests/test-class-amp-post-type-support.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * Tests for Post Types Support.
+ *
+ * @package AMP
+ * @since 0.6
+ */
+
+/**
+ * Tests for Post Type Support.
+ *
+ * @covers AMP_Post_Type_Support
+ */
+class Test_AMP_Post_Type_Support extends WP_UnitTestCase {
+
+	/**
+	 * Test add_hooks().
+	 *
+	 * @covers AMP_Post_Type_Support::init()
+	 */
+	public function test_init() {
+		AMP_Post_Type_Support::init();
+		$this->assertEquals( 5, has_action( 'after_setup_theme', array( 'AMP_Post_Type_Support', 'add_post_type_support' ) ) );
+	}
+
+	/**
+	 * Test get_builtin_supported_post_types.
+	 *
+	 * @covers AMP_Post_Type_Support::get_builtin_supported_post_types()
+	 */
+	public function test_get_builtin_supported_post_types() {
+		$this->assertEquals( array( 'post' ), AMP_Post_Type_Support::get_builtin_supported_post_types() );
+	}
+
+	/**
+	 * Test get_eligible_post_types.
+	 *
+	 * @covers AMP_Post_Type_Support::get_eligible_post_types()
+	 */
+	public function test_get_eligible_post_types() {
+		register_post_type( 'book', array(
```

Comment:
These post types are not removed on tear down which could affect other tests.

Reply:
Unnecessary because tests in core automatically clean up after themselves: https://github.com/WordPress/wordpress-develop/blob/1c2b0762c7f1a91394c633ae812505919bd297ed/tests/phpunit/includes/testcase.php#L102-L182

Reply:
I might be missing something but I don't see where the registered post types are removed, for instance if remove the post type registration in `test_add_post_type_support()` the tests still pass.

Reply:
I ended up explicitly unregistering the post types in 8c140631320654af9d2ba2316731feb61c69374d because it seems core isn't fully cleaning up after itself as I thought.